haskell/toolchain.bzl

```diff
@@ -105,11 +106,13 @@ def _haskell_toolchain_impl(ctx):
         if not tool in ghc_binaries:
             fail("Cannot find {} in {}".format(tool, ctx.attr.tools.label))
 
+    full_ghc = depset(transitive = [x.files for x in ctx.attr.full_ghc])
```

Comment:
Note, you can use [`ctx.files.full_ghc` as a short-cut](https://docs.bazel.build/versions/master/skylark/lib/ctx.html#files).